cmd/kyma/alpha/provision/k3s/cmd.go

```diff
@@ -114,12 +115,17 @@ func (c *command) deleteExistingK3sCluster() error {
 }
 
 //Check if a port is allocated
-func (c *command) portAllocated(port int) bool {
-	con, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
-	if con != nil {
-		con.Close()
+func (c *command) portAllocated(ports ...int) error {
+	for _, port := range ports {
+		con, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
+		if err != nil {
+			return err
+		}
+		if con != nil {
+			con.Close()
+		}
```

Comment:
In case of error connection would stay open!
```suggestion
		con, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
		defer con.Close()
		if err != nil {
			return err
		}
```